store/options.go

```diff
@@ -0,0 +1,44 @@
+package store
+
+// PruneOptions defines the pruning configuration.
+type PruneOptions struct {
+	// KeepRecent sets the number of recent versions to keep.
+	KeepRecent uint64
+
+	// Interval sets the number of how often to prune.
+	// If set to 0, no pruning will be done.
+	Interval uint64
+}
+
+// DefaultPruneOptions returns the default pruning options.
+// Interval is set to 0, which means no pruning will be done.
+func DefaultPruneOptions() *PruneOptions {
```

Comment:
This isn't in line with default pruning behavior specified in other locations and docs.  `app.toml` is parsed with the following in mind, see also https://docs.cosmos.network/v0.47/learn/advanced/config

```toml
# default: the last 362880 states are kept, pruning at 10 block intervals
# nothing: all historic states will be saved, nothing will be deleted (i.e. archiving node)
# everything: 2 latest states will be kept; pruning at 10 block intervals.
# custom: allow pruning options to be manually specified through 'pruning-keep-recent', and 'pruning-interval'
pruning = "default"

# These are applied if and only if the pruning strategy is custom.
pruning-keep-recent = "0"
pruning-interval = "0"
```

I think it's fine we leave interpretation of the magic strings "default", "nothing", "everything" to configuration higher up the stack, but default behavior of no pruning doesn't seem right to me.

Reply:
ok, that's a good point, btw This value should be tuned in the store/v2. I will create a new task to track it.

Reply:
https://github.com/cosmos/cosmos-sdk/issues/19421